requests.go

```diff
@@ -0,0 +1,294 @@
+	err = sendRequest(client, http.MethodPost, scheme+hostname+loginSubmitPath, headers, body, userData)
+	if err != nil {
+		return err
+	}
+
+	if userData.SessionToken == "" {
+		return errors.New("missing sessionToken in response body; user might be locked out of portal")
+	}
+
+	// Start the oauth2 process
+	// GET scheme+idmHostname+/login/sessionCookieRedirect?token=&redirectUrl=scheme+hostname+/myportal/
+	// get the sessionToken from the response body of the POST /portal/login request and use as oauth2 token
+	token := userData.SessionToken
+	params := fmt.Sprintf("token=%s&redirectUrl=%s%s%s", token, scheme, hostname, oauth2RedirectUrlPath)
+
+	headers = map[string][]string{
+		"Host":                      {idmHostname},
+		"upgrade-insecure-requests": {"1"},
+		"sec-fetch-site":            {"same-site"},
+		"sec-fetch-mode":            {"navigate"},
+		"sec-fetch-dest":            {"document"},
+		"sec-fetch-user":            {"?1"},
+		"referer":                   {scheme + hostname},
+		"origin":                    {hostname},
+	}
+	err = sendRequest(client, http.MethodGet, scheme+idmHostname+loginOauth2Path+params, headers, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func changePasswordStep(client *http.Client, portal *Portal, oldPassword, newPassword string) error {
+	// Get userID from response body of:
+	// GET https://idm.cms.gov/api/v1/sessions/me/lifecycle/refresh
+	hostname := portal.Hostname
+	idmHostname := portal.IDMHostname
+
+	headers := map[string][]string{
+		"Host":                       {idmHostname},
+		"x-okta-user-agent-extended": {"okta-auth-js-1.8.0"}, // brittle
+		"x-requested-with":           {"XMLHttpRequest"},
+		"sec-fetch-site":             {"same-site"},
+		"sec-fetch-mode":             {"cors"},
+		"sec-fetch-dest":             {"empty"},
+		"referer":                    {scheme + hostname + "/myportal/view-profile"},
+		"origin":                     {scheme + hostname},
+	}
+
+	userData := &userData{}
+	err := sendRequest(client, http.MethodPost, scheme+idmHostname+userDataPath, headers, nil, userData)
+	if err != nil {
+		return err
+	}
+
+	if userData.UserId == "" {
+		return errors.New("missing userId in response body")
+	}
+
+	// Change password: scheme+hostname+/myportal/viewprofile/myprofile/credential
+	creds := changePassword{
+		OldPassword: oldPassword,
+		NewPassword: newPassword,
+	}
+
+	body, err := json.Marshal(creds)
+	if err != nil {
+		return err
+	}
+
+	portalXsrfTokenCookie := getCookie(client, scheme+hostname, "PORTAL-XSRF-TOKEN")
+	headers = map[string][]string{
+		"Host":              {hostname},
+		"sec-fetch-site":    {"same-origin"},
+		"sec-fetch-mode":    {"cors"},
+		"sec-fetch-dest":    {"empty"},
+		"referer":           {scheme + hostname + "/myportal/view-profile"},
+		"origin":            {scheme + hostname},
+		"xhr_request":       {"true"},
+		"observe":           {"response"},
+		"portal-xsrf-token": {portalXsrfTokenCookie.Value},
```

Comment:
this crashes the program if the cookie doesn't come back

Reply:
getCookie() now returns cookie and error
The caller checks the error, logs it, and continues iterating.
This made me think that `changeUserPassword` should delete all cookies. I should probably move the defer() func to remove all cookies from `logoutStep` to `changeUserPassword`. What do you think?